packages/test/test-end-to-end-tests/src/test/summarizerWithLocalChanges.spec.ts

```diff
@@ -313,6 +321,120 @@ describeNoCompat("Summarizer with local data stores", (getTestObjectProvider) =>
 		},
 	);
 
+	itExpects(
+		"ValidateSummaryBeforeUpload = true. Summary should fail if ops are sent before summarize",
+		[
+			{
+				eventName: "fluid:telemetry:Summarizer:Running:Summarize_cancel",
+				clientType: "noninteractive/summarizer",
+				error: "PendingOpsWhileSummarizing",
+				beforeGenerate: true,
+			},
+		],
+		async () => {
+			// Wait for 100 ms for pending ops to be saved.
+			const pendingOpsTimeout = 100;
+			settings["Fluid.ContainerRuntime.SubmitSummary.waitForPendingOpsTimeout"] =
+				pendingOpsTimeout;
+			const mockLogger = new MockLogger();
+			const container1 = await provider.makeTestContainer();
+			const { summarizer, container: summarizerContainer } = await createSummarizer(
+				provider,
+				container1,
+				{ loaderProps: { configProvider: mockConfigProvider(settings) } },
+				undefined /* summaryVersion */,
+				mockLogger,
+			);
+
+			const defaultDataStore1 = await requestFluidObject<ITestDataObject>(
+				summarizerContainer,
+				"default",
+			);
+
+			// Pause op processing and send ops so there are pending ops in the summarizer.
+			const pendingOpCount = 10;
+			await provider.opProcessingController.pauseProcessing(summarizerContainer);
+			for (let i = 0; i < pendingOpCount; i++) {
+				defaultDataStore1._root.set(`key${i}`, `value${i}`);
+			}
+
+			await assert.rejects(
+				async () => {
+					await summarizeNow(summarizer);
+				},
+				(error: any) => {
+					// The summary should have failed because of "PendingOpsWhileSummarizing" error in "base" stage.
+					return (
+						error.message === "PendingOpsWhileSummarizing" &&
+						error.data.stage === "base"
+					);
+				},
+				"expected PendingOpsWhileSummarizing",
+			);
+
+			// We should have received a PendingOpsWhileSummarizing event with all the pending ops not saved.
+			mockLogger.assertMatch([
+				{
+					eventName: "fluid:telemetry:Summarizer:Running:PendingOpsWhileSummarizing",
+					saved: false,
+					countBefore: pendingOpCount,
+					countAfter: pendingOpCount,
```

Comment:
Put these in `details`?

Reply:
It's kinda hard to do this because not all sources of telemetry events support passing in details. For example, this event logs properties from [RetriableSummaryError](https://github.com/microsoft/FluidFramework/blob/74f413acc9a02e31c494a6cb3490d1684c9183a6/packages/runtime/container-runtime/src/summary/summaryGenerator.ts#L189) which does not accept it.
Also, I am not sure if mock logger supports comparing details type properties.

Reply:
k